app/src/components/TitledPanel.tsx

```diff
@@ -0,0 +1,196 @@
+import React, {
+  forwardRef,
+  PropsWithChildren,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from "react";
+import { flushSync } from "react-dom";
+import {
+  ImperativePanelHandle,
+  Panel,
+  PanelProps,
+  PanelResizeHandle,
+  PanelResizeHandleProps,
+} from "react-resizable-panels";
+import { css } from "@emotion/react";
+
+import { Icon, Icons } from "@phoenix/components";
+import { compactResizeHandleCSS } from "@phoenix/components/resize";
+
+/**
+ * A panel with a title that can be collapsed and expanded.
+ *
+ * Add `resizable` prop to make the panel resizable with an automatically added handle.
+ * The first TitledPanel in a PanelGroup SHOULD NOT be resizable or things will break.
+ */
+export const TitledPanel = forwardRef<
+  ImperativePanelHandle | null,
+  PropsWithChildren<{
+    title: React.ReactNode;
+    panelProps?: PanelProps;
+    panelResizeHandleProps?: PanelResizeHandleProps;
+    resizable?: boolean;
+    bordered?: boolean;
+    expandSizeFallback?: number;
+    fullSize?: boolean;
+  }>
+>(
+  (
+    {
+      children,
+      title,
+      panelProps,
+      panelResizeHandleProps,
+      fullSize = false,
+      resizable = false,
+      bordered = true,
+    },
+    ref
+  ) => {
+    const panelRef = useRef<ImperativePanelHandle | null>(null);
+    useImperativeHandle<
+      ImperativePanelHandle | null,
+      ImperativePanelHandle | null
+    >(ref, () => panelRef.current);
+    const [collapsed, setCollapsed] = useState(false);
+
+    const handleClick = () => {
+      const panel = panelRef.current;
+      if (panel?.isCollapsed()) {
+        setCollapsed(false);
+        // if the panel is collapsed when its size is below 35%, it will be expanded to at least 35%
+        // otherwise it will return to its pre-collapse size
+        panel?.expand(fullSize ? 100 : 35);
+      } else {
+        setCollapsed(true);
+        panel?.collapse();
+      }
+    };
+
+    return (
+      <>
+        {resizable && (
+          <PanelResizeHandle
+            {...panelResizeHandleProps}
+            data-bordered={bordered}
+            css={css(
+              compactResizeHandleCSS,
+              css`
+                border-radius: var(--ac-global-rounding-small);
+                opacity: 1;
+                background-color: unset;
+                &[data-bordered="true"] {
+                  background-color: var(--ac-global-border-color-default);
+                }
+                &[data-panel-group-direction="vertical"] {
+                  height: 1px;
+                }
+                &:hover,
+                &:focus,
+                &:active,
+                &:focus-visible {
+                  // Make hover target bigger
+                  background-color: var(--ac-global-color-primary);
+                }
+                &:not([data-resize-handle-state="drag"]) ~ [data-panel] {
+                  // transition: flex 0.2s ease-in-out;
+                }
```

Comment:
probably could be moved into it's own thinResizeHandleCSS rather than composing styles